File: 백준/Silver/4158. CD/CD.py
import sys
input = sys.stdin.readline
while True:
    n,m = map(int,input().split())
    if n==0 and m==0:
        break
    sang = list(int(input()) for _ in range(n))
    sun = list(int(input()) for _ in range(m))
    idx_n, idx_m = 0,0
    cnt = 0
    while idx_n<n and idx_m<m:
        if sang[idx_n]==sun[idx_m]:
            cnt +=1
            idx_n +=1
            idx_m +=1
        elif sang[idx_n]<sun[idx_m]:
            idx_n+=1
        else:
            idx_m+=1
    print(cnt)

File: 백준/Silver/4158. CD/CD.java
import java.io.*;
import java.util.*;

public class Main {
    static int n;
    static int[][] graph; 
    static boolean[] visited;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
       while(true){
            String[] input = br.readLine().split(" ");
            int n = Integer.parseInt(input[0]);
            int m = Integer.parseInt(input[1]);

            if(n==0 && m==0){
                break;
            }
            List<Integer> sang = new ArrayList<>();
            List<Integer> sun  = new ArrayList<>();

            for(int i=0;i<n;i++){
                sang.add(Integer.parseInt(br.readLine()));
            }
            for(int i=0;i<m;i++){
                sun.add(Integer.parseInt(br.readLine()));
            }
            int idxN = 0, idxM = 0;
            int cnt = 0;

            while(idxN< n && idxM<m){
                int sangCD = sang.get(idxN);
                int sunCD = sun.get(idxM);

                if(sangCD == sunCD){
                    cnt++;
                    idxM++;
                    idxN++;
                }else if(sangCD<sunCD){
                    idxN++;
                }else{
                    idxM++;
                }
            }
            System.out.println(cnt);
       }
}
}
